### PR TITLE
Fixed reset session button state

### PR DESCRIPTION
**Pull request by bennyn**

### CODE REVIEW

File: app/script/event/EventRepository.coffee [hunk 1]
```diff
@@ -355,14 +355,11 @@ class z.event.EventRepository
         return @cryptography_repository.decrypt_event event
         .catch (decrypt_error) =>
           # Get error information
+          error_code = decrypt_error.code
```
Comment (gregor) [inline app/script/event/EventRepository.coffee:358]: Make sure that it is not overwritten anywhere - L375 for example

**Comment by herrmannplatz** [comment]
Check _map_system_event_unable_to_decrypt in EventMapper

**Comment by bennycode** [comment]
Error record in database:

```javascript
{
  "category": 0,
  "conversation": "fb1c051a-3ce3-46c5-bbc2-0153b6076af0",
  "error": "We received a message with session tag 'a8859a310a0c374a3da67e3a0f871145', but we don't have a session for this tag. (c0a70d96aaeb87b6)",
  "error_code": "205 (c0a70d96aaeb87b6)",
  "from": "2bde49aa-bdb5-458f-98cf-7d3552b10916",
  "id": "cb4972e0-9586-42a2-90cc-1798ec0cb648",
  "primary_key": 9,
  "time": "2017-04-03T12:58:04.301Z",
  "type": "conversation.unable-to-decrypt"
}
```

File: app/script/event/EventRepository.coffee (hunk 1 above)
Comment (bennycode) [reply app/script/event/EventRepository.coffee:358]: Should be ok now? 😃

File: app/script/conversation/EventMapper.coffee [hunk 2]
```diff
@@ -478,9 +478,9 @@ class z.conversation.EventMapper
   ###
   _map_system_event_unable_to_decrypt: (event) ->
     message_et = new z.entity.DecryptErrorMessage()
-    # error_code style "3690 (f0c0272e8f053774)"
-    message_et.error_code = event.error_code?.substring(0, 4)
-    message_et.client_id = event.error_code?.substring(5).replace(/[()]/g, '')
+    if event.error_code
```
Comment (herrmannplatz) [inline app/script/conversation/EventMapper.coffee]: we dont need the if

**Comment by bennycode** [comment]
**Error lifecycle:**

`EventRepository._handle_event -> EventBuilder.build_unable_to_decrypt -> EventMapper._map_system_event_unable_to_decrypt -> DecryptErrorMessage`

**Recoverable error states:**

```javascript
z.cryptography.CryptographyErrorType.DUPLICATE_MESSAGE // 209
z.cryptography.CryptographyErrorType.INVALID_MESSAGE_SESSION_NOT_MATCHING // 
z.cryptography.CryptographyErrorType.INVALID_MESSAGE_SESSION_MISSING // 
z.cryptography.CryptographyErrorType.INVALID_SIGNATURE // 206, 207
z.cryptography.CryptographyErrorType.OUTDATED_MESSAGE // 208
z.cryptography.CryptographyErrorType.PRE_KEY_NOT_FOUND // 203
z.cryptography.CryptographyErrorType.TOO_DISTANT_FUTURE // 211
```

**Pull request merged**